functions/Export-DbaDacpac.ps1

```diff
@@ -0,0 +1,179 @@
+
+	.PARAMETER SqlCredential
+	Allows you to login to servers using SQL Logins instead of Windows Authentication (AKA Integrated or Trusted). To use:
+
+	$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter.
+
+	Windows Authentication will be used if SqlCredential is not specified. SQL Server does not accept Windows credentials being passed as credentials.
+
+	To connect as a different Windows user, run PowerShell as that user.
+
+	.PARAMETER Directory
+	The directory where the .dacpac files will be exported to. Defaults to documents.
+	
+	.PARAMETER Database
+	The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+	.PARAMETER ExcludeDatabase
+	The database(s) to exclude - this list is auto-populated from the server
+
+	.PARAMETER AllUserDatabases
+	Run command against all user databases
+	
+	.PARAMETER ExtendedParameters
+	Optional parameters used to extract the DACPAC. More information can be found at
+	https://msdn.microsoft.com/en-us/library/hh550080.aspx
+	
+	.PARAMETER ExtendedProperties
+	Optional properties used to extract the DACPAC. More information can be found at
+	https://msdn.microsoft.com/en-us/library/hh550080.aspx
+	
+
+	.PARAMETER EnableException
+	By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+	This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+	Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+	.NOTES
+	Tags: migration, database 
+	Author: Richie lee (@bzzzt_io)
+
+	Website: https://dbatools.io
+	Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+	License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+	
+	.LINK
+	https://dbatools.io/Export-DbaDacpac
+
+	.EXAMPLE
+	Export-DbaDacpac -SqlInstance sql2016 -Database SharePoint_Config
+	Exports the dacpac for SharePoint_Config on sql2016 to $home\Documents\SharePoint_Config.dacpac
+
+	.EXAMPLE
+	$moreprops = "/p:VerifyExtraction=$true /p:CommandTimeOut=10"
+	Export-DbaDacpac -SqlInstance sql2016 -Database SharePoint_Config -Directory C:\temp -ExtendedProperties $moreprops
+
+	Sets the CommandTimeout to 10 then extracts the dacpac for SharePoint_Config on sql2016 to C:\temp\SharePoint_Config.dacpac then verifies extraction.
+
+	
+    #>
+	[CmdletBinding()]
+	param
+	(
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[object[]]$Database,
+		[object[]]$ExcludeDatabase,
+		[switch]$AllUserDatabases,
+		[string]$Directory = "$home\Documents",
+		[string]$ExtendedParameters,
+		[string]$ExtendedProperties,
+		[switch]$EnableException
+	)
+	
+	process {
+		if (!$Database -and !$ExcludeDatabase -and !$AllUserDatabases) {
```

Comment:
Didn't we replace such checks with `Test-Bound` or `$PSBoundParameters`?

Reply:
Yes, I forgot, it is Test-Bound, will update thanks